hack/verify-manifests.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+DIFFROOT="${REPO_ROOT}/config"
+TMP_ROOT="${REPO_ROOT}/_tmp"
+
+
+cleanup() {
+    rm -rf "${TMP_ROOT}"
+}
+trap "cleanup" EXIT SIGINT
+cleanup
+
+MANIFEST_ROOT=$TMP_ROOT make manifests
+
+echo "diffing ${DIFFROOT} against freshly generated manifests"
+ret=0
+git diff --no-index --diff-filter=MD ${TMP_ROOT} ${DIFFROOT} > /dev/null 2>&1 || ret=$?
```

Comment:
I am using `git diff` rather than `diff` here because the output is cleaner. I can't get `diff` to ignore files not present in the `TMP_ROOT`. There are many files in `DIFFROOT` that don't seem to be generated files.

Reply:
`git diff` will not see newly created files, so this test won't be 100% sufficient

Reply:
This test...
- generates expected manifest files in `_tmp`
- `git diff` will...
  - see if `/config` has all the files generated in `_tmp`
  - diff the matching `/config` and `_tmp` files

In which situation do we need to see newly created files in `config`.

Reply:
This test will...
- run `make manifests` and output the results into `_tmp`
- `git diff` will...
  - see if there exists files in `_tmp` that don't exist in `config`
  - see if the matching files in `_tmp` and `config` are the same

If by "newly created files" you mean files that exist in `config` and not in `_tmp`, there seemed to be many files in `config` that weren't generated.

Can you elaborate.

Reply:
Nevermind, I didn't see this was using `--no-index`. My bad!

Reply:
dope. Thanks for reviewing!